app/assets/javascripts/student_profile/TransitionNotes.js

```diff
@@ -54,52 +54,17 @@ class TransitionNotes extends React.Component {
     this.state = {
       noteText: (regularNote ? regularNote.text : notePrompts),
       restrictedNoteText: (restrictedNote ? restrictedNote.text : restrictedNotePrompts),
-      regularNoteAutosaveStatus: null,
-      restrictedNoteAutosaveStatus: null,
     };
 
-    const throttleOptions = {leading: true, trailing: true};
+    const throttleOptions = {leading: false, trailing: true};
 
-    this.autosaveRegularNote = _.throttle(this.autosaveRegularNote.bind(this), 3000, throttleOptions);
-    this.autosaveRestrictedNote = _.throttle(this.autosaveRestrictedNote.bind(this), 3000, throttleOptions);
+    this.autosaveRegularNote = _.throttle(this.autosaveRegularNote.bind(this), 1000, throttleOptions);
+    this.autosaveRestrictedNote = _.throttle(this.autosaveRestrictedNote.bind(this), 1000, throttleOptions);
 
     this.onChangeRegularNote = this.onChangeRegularNote.bind(this);
     this.onChangeRestrictedNote = this.onChangeRestrictedNote.bind(this);
   }
 
-  // Note on componentWillReceiveProps:
-  // This method will be deprecated in React 16.3. But we're still using
-  // React 15.4 in Insights, and have some legacy-cleanup work to finish
-  // til we can migrate to 16. Noting that this func will have to be refactored.
-
-  // When the user types into the Transition Note textarea, the server
-  // handles autosaving very quickly. In fact, so fast that in the "pending" state,
-  // the status text is barely readable when it's supposed to say "Autosaving..."
-
-  // This code adds a delay so that the text that says "Autosaving..."
-  // can hang for  2 seconds to let the user read it.
-  componentWillReceiveProps(newProps) {
```

Comment:
Another thing that might be good to add is a warning when the user tries to close the page or navigate away, but the UI thinks there are unsaved changes.  There's a browser hook for this called `beforeunload` so this is easy to add, here's an example you could check out: https://github.com/studentinsights/studentinsights/blob/584775c509f94d433cb06c1ef769e472a02e0c40/app/assets/javascripts/class_lists/ClassListCreatorPage.js#L112